node/service/src/grandpa_support.rs

```diff
@@ -176,8 +176,10 @@ impl<B> grandpa::VotingRule<PolkadotBlock, B> for ApprovalCheckingVotingRule
 				let diff = best_number.saturating_sub(base_number);
 				if diff >= MAX_APPROVAL_CHECKING_FINALITY_LAG {
 					// Catch up to the best, with some extra lag.
-					find_target(best_number - MAX_APPROVAL_CHECKING_FINALITY_LAG, &best_header)
-						.filter(|vote| vote.1 <= current_number)
+					match find_target(best_number - MAX_APPROVAL_CHECKING_FINALITY_LAG, &best_header) {
```

Comment:
If we don't find the target or if the given target is higher than current then we should stick with current.

Reply:
`find_target` returns `None` only when db/IO error happens, no? TBH it should rather return a `Result`

Reply:
It can also return `None` if the given header isn't found (which should not happen).

Reply:
Done.